foo2zjs/foo2zjs-pstops.1
'\" t
'\"
'\"	DO NOT EDIT! This file is generated from foo2zjs-pstops.1in
'\"
.TH foo2zjs-pstops 1 "Fri Apr 01 11:12:13 2011" "foo2zjs-pstops 0.0"
'\"
'\"
'\"==========================================================================
'\"	STRINGS and MACROS
'\"==========================================================================
'\"
'\"	Define strings for special characters that nroff doesn't have
'\"
'\"	N.B. using .if cua to test for special characters did not work.
'\"
.if !'\*[.T]'ps' .ds ua up
.if  '\*[.T]'ps' .ds ua \(ua
.if !'\*[.T]'ps' .ds da down
.if  '\*[.T]'ps' .ds da \(da
.if !'\*[.T]'ps' .ds <- left
.if  '\*[.T]'ps' .ds <- \(<-
.if !'\*[.T]'ps' .ds -> right
.if  '\*[.T]'ps' .ds -> \(->
'\"
'\"     bx - box a Courier string for making keycaps
'\"
'\"	N.B. this mess is to make the line drawing come out only
'\"	if we are really generating postscript
'\"
.de bx
.ie !'\*[.T]'ps' \{\
.	RB [ \\$1 ]\\$2
.\}
.el \{\
.	ie !r ps4html \{\
'\" \(br\|\s-1\f(CB\\$1\fP\s+1\|\(br\l'|0\(rn'\l'|0\(ul'
.		ft CW
.		nr par*bxw \w'\\$1'+.4m 
\Z'\v'.25m'\D'l 0 -1m'\D'l \\n[par*bxw]u 0'\D'l 0 1m'\D'l -\\n[par*bxw]u 0''\
\Z'\h'.2m'\s-1\\$1\s+1'\
\h'\\n[par*bxw]u'\\$2
.		ft P
.	\}
.	el \{\
.		RB [ \\$1 ]\\$2
.	\}
.\}
..
'\"
'\" strings to set current color (null with old groff)
'\"
.if mred .ds red \m[red]
.if mgreen .ds green \m[green]
.if mblue .ds blue \m[blue]
.if mblack .ds black \m[black]
.if mblack .ds mP \mP
'\"
'\" fix for grotty + xterm. We call for orange, grotty outputs yellow,
'\" but xterm displays yellow as orange.  The cycle is complete.
'\"
.if n .defcolor orange rgb #ffff00
'\"
'\" color <color> - set the current color (ignores request with old groff)
'\"
.de color
.if mred \m[\\$1]\c
..
'\"
'\" colorword <color> <word> - colorize a word (ignored by old groff)
'\"
.de colorword
.ie m\\$1 \m[\\$1]\\$2\mP\c
.el \\$2\c
..
'\"
'\" colbox <fg> <bg> <word> - colorize a word in a filled box
'\"
.de colbox
.ie mred \M[\\$2]\
\v'+.167v'\
\D'P 0 -0.9v  \w'\\$3'u 0  0 +0.9v   -\w'\\$3'u 0'\
\v'-.167v'\
\m[\\$1]\\$3\mP\MP
.el \\$3\c
..
'\"
'\"	Macros for doing pdfmarks
'\"
.de specialps
.if  '\*[.T]'ps' \\k_\X'ps: \\$*'\h'|\\n_u'\c
..
'\"
'\" pdfmark PDFMARKCODE
'\"
.ds pdfmarks
.if d pdfmarks \{\
.de pdfmark
.	specialps exec [\\$1 pdfmark
..
'\"
'\" pdfdest LINKNAME
'\"
.de pdfdest
.pdfmark "/Dest /\\$1 /View [/XYZ -5 PL null] /DEST"
..
'\"
'\" pdfbookmark COUNT LINKNAME STRING
'\"
.de pdfbookmark
.   pdfmark "/View [/XYZ 44 730 1.0] /Count \\$1 /Dest /\\$2 /Title (\\$3) /OUT"
..
'\"
'\"	Define the SH and SS macros to save pdfmark information
'\"	in "arrays" of numbers and strings.
'\"
.if !r rr_n \{\
.nr rr_n 0 1
.am SH
.	nr rr_levels!\\n+[rr_n] 2
.	ds rr_labels!\\n[rr_n] \\$*
.	pdfdest Link\\n[rr_n]
..
.am SS
.	nr rr_levels!\\n+[rr_n] 3
.	ds rr_labels!\\n[rr_n] \\$*
.	pdfdest Link\\n[rr_n]
..
.\}
'\"
'\"	Called at the end of the document to generate the pdfmark outline
'\"
.de pdf_outline
.nr rr_levels!\\n+[rr_n] 1
.nr rr_i 0 1
.while \\n+[rr_i]<\\n[rr_n] \{\
.   nr rr_ip1 \\n[rr_i]+1
.   nr rr_count 0
.   if \\n[rr_levels!\\n[rr_ip1]]>\\n[rr_levels!\\n[rr_i]] \{\
.       nr rr_j \\n[rr_i] 1
.       while \\n+[rr_j]<\\n[rr_n] \{\
.           if \\n[rr_levels!\\n[rr_j]]<=\\n[rr_levels!\\n[rr_i]] \{\
.               break
.           \}
.           if \\n[rr_levels!\\n[rr_j]]==(\\n[rr_levels!\\n[rr_i]]+1) \{\
.               nr rr_count \\n[rr_count]+1
.           \}
.       \}
.   \}
.   ds hhh \\*[rr_labels!\\n[rr_i]]
.   pdfbookmark -\\n[rr_count] Link\\n[rr_i] "\\*[hhh]"
.\}
..
'\"
'\" Some postscript to make pdfmarks harmless on old interpreters...
'\"
.specialps "def /pdfmark where {pop} {userdict /pdfmark /cleartomark load put} ifelse"
'\"
'\" Force display of Bookmarks in Acrobat when document is viewed.
'\"
.pdfmark "[/PageMode /UseOutlines /Page 1 /View [/XYZ null null null] /DOCVIEW"
'\"
'\" Output the document info in pdfmarks
'\"
.pdfmark "\
	/Title (\*[an-title](\*[an-section])) \
	/Subject (\*[an-title] Manual Page) \
	/Author (Rick Richardson) \
	/Keywords (printing) \
	/Creator (groff \n(.x.\n(.y.\n(.Y -man) \
	/CreationDate (\*[an-extra1]) \
	/ModDate (\*[an-extra1]) \
	/DOCINFO"
\}
'\"
'\" The manual page name is only 1st level mark
'\"
.nr rr_levels!\n+[rr_n] 1
.ds rr_labels!\n[rr_n] \*[an-title](\*[an-section])
.pdfdest Link\n[rr_n]
'\"
'\"==========================================================================
'\"	MANUAL PAGE SOURCE
'\"==========================================================================
.SH NAME
foo2zjs-pstops \- Add PS code for foo2*-wrapper
.SH SYNOPSIS
.B foo2zjs-pstops
.RI [ options "] [" file ]
.SH DESCRIPTION
Add PS code for foo2zjs-wrapper.

.SH COMMAND LINE OPTIONS
These are the options that can appear on the command line.
.TP
.BI \-a
Accurate screens code.
.TP
.BI \-c
CIE Color.
.TP
.BI \-n
Neuter CUPS cupsPSLevel2
.TP
.BI \-r
Rotate 90 degrees clockwise.
.TP
.BI \-w
Well Tempered Screens code.
.TP
.BI \-D\0 level
Set Debug level [0].

.SH FILES
.BR /usr/bin/foo2zjs-pstops
.SH SEE ALSO
.BR foo2hp2600-wrapper (1),
.BR foo2lava-wrapper (1),
.BR foo2oak-wrapper (1),
.BR foo2qpdl-wrapper (1),
.BR foo2slx-wrapper (1),
.BR foo2xqx-wrapper (1),
.BR foo2zjs-wrapper (1)
.SH "AUTHOR"
Rick Richardson <rick.richardson@comcast.net>
.br
http://foo2zjs.rkkda.com/
'/"
'/"
'/"
.em pdf_outline


foo2zjs/ChangeLog
2011-03-08	Rick Richardson <rick.richardson@comcast.net>
	* hplj10xx.conf: For FreeBSD 8, s/zjx/xqx/ for HP LJ P???? printers
	    Ref: Yury G. Kudryashov <urkud.urkud@gmail.com>

2011-02-10	Rick Richardson <rick.richardson@comcast.net>
	* Makefile: fix perms on PPD files.

2011-01-31	Rick Richardson <rick.richardson@comcast.net>
	* New Printer: Oki C310dn.

2011-01-23	Rick Richardson <rick.richardson@comcast.net>
	* osx-hotplug/osx-hplj-hotplug.1in: Man page

2011-01-21	Rick Richardson <rick.richardson@comcast.net>
	* osx-hotplug/osx-hplj-hotplug.m:
	  If FWVER is present, then no download is needed.

2011-01-20	Rick Richardson <rick.richardson@comcast.net>
	* Makefile: make install-hotplug for OSX.

2011-01-19	Rick Richardson <rick.richardson@comcast.net>
	* Change foo2hp2600-wrapper to remove "expr substr ..." and
	  replace it with "echo ... | sed ..." for OSX (because expr != gexpr)

2011-01-18	Rick Richardson <rick.richardson@comcast.net>
	* New program: osx-hplj-hotplug
	    Does hotplug on Mac OSX.  Add a line to /etc/rc.local and reboot:

	    killall osx-hplj-hotplug; osx-hplj-hotplug \
		    >/tmp/osx-hplj-hotplug 2>&1 &

2011-01-14	Rick Richardson <rick.richardson@comcast.net>
	* foo2*-wrapper, foomatic-db, PPDs: Add Halftone Algoritm.
	    Standard (QUALITY=1), Accurate Screens (QUALITY=as),
	    Well Tempered Screens (QUALITY=wts).  WTS is the default.
	    PLEASE delete and recreate ALL printers.

2011-01-09	Rick Richardson <rick.richardson@comcast.net>
	* foo2*-wrapper:
	    use QUALITY=1 instead of QUALITY=wts due to a bug in gs (8.71+)

2011-01-09	Rick Richardson <rick.richardson@comcast.net>
	* foomatic-db, PPDs: Change ICM value for CLX-3175 (comestic)

2010-12-22	Rick Richardson <rick.richardson@comcast.net>
	* New printers: HP LaserJet 1022n, HP LaserJet 1022nw
	* hplj1020.desktop: change Name and Comment

2010-12-13	Rick Richardson <rick.richardson@comcast.net>
	* foo2zjs.c: with -z3, use BIH width for ZJI_RASTER_X
	  Tested with a real HP LaserJet Pro CP1025nw printer.

	* ICM for -z3 is done.  Do: make; ./getweb 1025; make install
	  and PLEASE delete and recreate the foo2zjs-z3 printers

2010-12-12	Rick Richardson <rick.richardson@comcast.net>
	* foo2zjs.c: with -z3, adjust to mod(128) pixels.

2010-12-10	Rick Richardson <rick.richardson@comcast.net>
	* Makefile, hplj1000 et al:
	    rm /etc/udev/rules.d/*hpmud*laserjet_{1000,1005,1018, 1020,
			p1005,p1006,p1007,p1008,p1505}*

	    Needed for Ubuntu 10.04 and greater with the proprietary "hplip"
	    package installed.

	    rm /etc/udev/rules.d/*hpmud_support.rules

	    Needed for Fedora 13+

2010-12-09	Rick Richardson <rick.richardson@comcast.net>
	* foo2zjs.c, foo2zjs-wrapper.in, zjsdecode, ppd/*, foomatic-db/*:
	    Add 4x6, 5x8, and 10x15 page size for HP LaserJet Pro CP1025nw

2010-12-08	Rick Richardson <rick.richardson@comcast.net>
	* foo2zjs.c, foo2zjs-wrapper.in, zjsdecode:
	    Add -z3 for HP LaserJet Pro CP1025nw
	* New Printer: HP LaserJet Pro CP1025nw
	    ALPHA quality!!!  No ICM for color yet!!!

2010-11-12	Rick Richardson <rick.richardson@comcast.net>
	* New Printer: Olivetti d-Color P160W

2010-11-02	Rick Richardson <rick.richardson@comcast.net>
	* foo2oak-wrapper: Change default HP CLJ 1500 ICM file. Do:
			make; ./getweb 1500; make install

2010-10-26	Rick Richardson <rick.richardson@comcast.net>
	* foo2*wrapper: Typo in GAMMAFILE

2010-10-26	József CZOMPÓ <czo@czo.hu>
	* foo2oak et al:  Add tray 2.
	* foo2oak.c:  make -A and -B compatible with foo2hp, foo2zjs.

2010-10-16	Rick Richardson <rick.richardson@comcast.net>
	* foo2lava.c: Iff MODE_COLOR and nbie = 1, then it is a mono page.

2010-10-12	Rick Richardson <rick.richardson@comcast.net>
	* getweb: move dl2300.tar.gz

2010-08-17	Rick Richardson <rick.richardson@comcast.net>
	* New foo2lava printer: Xerox Phaser 6121MFP (printer only)

2010-08-09	Rick Richardson <rick.richardson@comcast.net>
	* Makefile: only do command2foo2lava-pjl IFF cups-devel package
	    is installed

2010-08-07	Rick Richardson <rick.richardson@comcast.net>
	* foo2qpdl: -z3 resolution

2010-07-28	Rick Richardson <rick.richardson@comcast.net>
	* Added manual page for foo2zjs-icc2ps.

2010-07-26	Rick Richardson <rick.richardson@comcast.net>
	* command2foo2lava-pjl.c: switched to use foo2zjsNegateMarkerLevels

2010-07-25	Rick Richardson <rick.richardson@comcast.net>
	* command2foo2lava-pjl.c: improved formatting

2010-07-24	Rick Richardson <rick.richardson@comcast.net>
	* foo2*-wrapper: add eval gs.foo for BSD

2010-07-22	Rick Richardson <rick.richardson@comcast.net>
	* Modify "modify-ppd" file for *cupsSNMPSupplies: False
	* Modify: foo2zjs for Duplex and -z2 (HP P1606dn)

2010-07-20	Rick Richardson <rick.richardson@comcast.net>
	* Reinhold Kainhofer <reinhold@kainhofer.com>
	    * New file: command2foo2lava-pjl.c
	    * New file: modify-ppd
	    * Change PPD's for Konica Minolta mc1600W, mc1680MF, mc1690MF,
	      mc2490 MF, mc2530 DL, mc4690MF, and Oki C110 if cups-devel
	      is installed.
	    Used for reporting marker (toner) levels via PJL on
	    foo2lava printers.
	* command2foo2lava-pjl.c: Add a sleep(5) for mc2530 DL
	* command2foo2lava-pjl.c: Add IMGDRUM, CONSUMETONERCOUNTERFEIT*
	* command2foo2lava-pjl.c: Read PPD file for product

2010-06-20	Rick Richardson <rick.richardson@comcast.net>
	* foo2zjs, foo2zjs-wrapper, foomatic and ppd file:
	    Add Printer Density.  For HP zjs printers.  Please remove
	    and recreate the printers.

2010-06-19	Rick Richardson <rick.richardson@comcast.net>
	* Add manual page for gipddecode.
	* foo2xqx, foo2xqx-wrapper, foomatic and ppd file:
	    Add Printer Density.  For HP xqx printers.  Please remove
	    and recreate the printers.

2010-06-15	Rick Richardson <rick.richardson@comcast.net>
	* foo2qpdl foomatic-db and PPD files: fix Duplex for clp-6100, clp-610,
	  clp-620. Manual Duplex is no longer supported. Use Gnome-Manual-Duplex
	  instead.

	  Please delete and recreate the Samsung CLP-6?? printers.


foo2zjs/foo2zjs-pstops.sh
#!/bin/sh

VERSION='$Id: foo2zjs-pstops.sh,v 1.15 2011/01/13 17:48:29 rick Exp $'

PROGNAME="$0"

usage() {
	cat <<EOF
NAME
    `basename $PROGNAME` - Add PS code for foo2*-wrapper

SYNOPSIS
    `basename $PROGNAME` [options] [file]

DESCRIPTION
    Add PS code for foo2zjs-wrapper.

OPTIONS
    -h ydimpts	For rotate -r, y dimension points
    -r		Rotate 90 clockwise
    -a		Accurate Screens code
    -c		CIEColor
    -n		Neuter CUPS cupsPSLevel2
    -w		Well Tempered Screens code
    -D lvl	Debug level
EOF

	exit 1
}

#
#       Report an error and exit
#
error() {
	echo "`basename $PROGNAME`: $1" >&2
	exit 1
}

debug() {
	if [ $DEBUG -ge $1 ]; then
	    echo "`basename $PROGNAME`: $2" >&2
	fi
}

#
#       Process the options
#
DEBUG=0
ROTATE90=0
ACCURATE=0
CIECOLOR=0
NIXCUPS=0
WTS=0
while getopts "ach:nwrD:Vh?" opt
do
	case $opt in
	a)	ACCURATE=1;;
	c)	CIECOLOR=1;;
	h)	YDIMpoints="$OPTARG";;
	n)	NIXCUPS=1;;
	r)	ROTATE90=1;;
	w)	WTS=1;;
	D)	DEBUG="$OPTARG";;
	V)	echo "$VERSION"; exit 0;;
	h|\?)	usage;;
	esac
done
shift `expr $OPTIND - 1`

if [ $NIXCUPS = 1 ]; then
    n='s#^[^/]*cupsPSLevel2#false#'
else
    n=
fi

if [ $ROTATE90 = 1 ]; then
	    # %%currentpagedevice /PageSize get\
	    # %%aload pop translate\
    r="1i\
	<< /Install {\
	    0 $YDIMpoints translate\
	    -90 rotate\
	} >> setpagedevice
	"
else
    r=
fi

if [ $WTS = 1 ]; then
    w='/%%Page:.*[ 	]1[ 	]*$/ i\
	<< /UseWTS true >> setuserparams \
	<<\
	    /AccurateScreens true\
	    /HalftoneType 1\
	    /HalftoneName (Round Dot Screen) cvn\
	    /SpotFunction { 180 mul cos exch 180 mul cos add 2 div}\
	    /Frequency 137\
	    /Angle 37\
	>> sethalftone
        '
elif [ $ACCURATE = 1 ]; then
    w='/%%Page:.*[      ]1[     ]*$/ i\
	<< /UseWTS false >> setuserparams \
	<<\
	    /AccurateScreens true\
	    /HalftoneType 1\
	    /HalftoneName (Round Dot Screen) cvn\
	    /SpotFunction { 180 mul cos exch 180 mul cos add 2 div}\
	    /Frequency 137\
	    /Angle 37\
	>> sethalftone
	'
else
    w=
fi

#
#	Main Program
#
sed \
    -e "$w" \
    -e "$n" \
    -e "$r" \
    $@
